test.sh

```diff
@@ -64,10 +109,128 @@ function run_test_coverage() {
   goveralls -v -coverprofile=gover.coverprofile -service=travis-pro
 }
 
+#
+# Main CLI Parser
+#
+USAGE="$(cat -- <<-EOM
+
+Usage:
+Boulder test suite CLI, intended to be run inside of a Docker container:
+
+  docker-compose run --use-aliases boulder ./$(basename "${0}") [OPTION]...
+
+With no options passed, runs standard battery of tests (lint, unit, and integation)
+
+    -l, --lints                           Adds lint to the list of tests to run
+    -u, --unit                            Adds unit to the list of tests to run
+    -p <DIR>, --unit-test-package=<DIR>   Run unit tests for specific go package(s)
+    -e, --enable-race-detection           Enables -race flag for all unit and integration tests
+    -n, --config-next                     Changes BOULDER_CONFIG_DIR from test/config to test/config-next
+    -c, --coverage                        Adds coverage to the list of tests to run
+    -i, --integration                     Adds integration to the list of tests to run
+    -s, --start-py                        Adds start to the list of tests to run
+    -v, --gomod-vendor                    Adds gomod-vendor to the list of tests to run
+    -g, --generate                        Adds generate to the list of tests to run
+    -r, --rpm                             Adds rpm to the list of tests to run
+    -o, --list-integration-tests          Outputs a list of the available integration tests
+    -f <REGEX>, --filter=<REGEX>          Run only those tests matching the regular expression
+
+                                          Note:
+                                           This option disables the '"back in time"' integration test setup
+
+                                           For tests, the regular expression is split by unbracketed slash (/)
+                                           characters into a sequence of regular expressions
+
+                                          Example:
+                                           TestAkamaiPurgerDrainQueueFails/TestWFECORS
+    -h, --help                            Shows this help message
+
+EOM
+)"
+
+while getopts lueciosvgrnhp:f:-: OPT; do
+  if [ "$OPT" = - ]; then     # long option: reformulate OPT and OPTARG
+    OPT="${OPTARG%%=*}"       # extract long option name
+    OPTARG="${OPTARG#$OPT}"   # extract long option argument (may be empty)
+    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+  fi
+  case "$OPT" in
+    l | lints )                      RUN+=("lints") ;;
+    u | unit )                       RUN+=("unit") ;;
+    p | unit-test-package )          check_arg; UNIT_PACKAGES+=("${OPTARG}") ;;
+    e | enable-race-detection )      RACE="true" && export RACE  ;;
```

Comment:
Let's unconditionally export RACE when it's first defined. That will cause update to it to be exported to.

Alternately, we could leave `RACE` undefined until we get the flag, and when we get the flag do `export RACE=true`.

The way it is here could confuse a future editor, since actions inside the script would see either `RACE=false` or `RACE=true`, while actions in forked programs would see either `<nothing>` or `RACE=true`.